packages/model-viewer/src/features/scene-graph/sampler.ts

```diff
@@ -113,12 +114,12 @@ export class Sampler extends ThreeDOMElement implements SamplerInterface {
     return this[$threeTexture].rotation;
   }
 
-  get scale(): Vector2 {
-    return this[$threeTexture].repeat;
+  get scale(): Vector2DInterface {
+    return toVector2D(this[$threeTexture].repeat);
   }
 
-  get offset(): Vector2|null {
-    return this[$threeTexture].offset;
+  get offset(): Vector2DInterface|null {
```

Comment:
nit: unlike set, here you actually get a real `Vector2D`.

Reply:
do we want the set and get to have different parameter/return value type?

Reply:
Technically that's the most accurate - we give a specific object back, but accept a broader interface class as input.